packages/next-app/src/contexts/InternalManagerContext.tsx

```diff
@@ -44,12 +61,31 @@ export function InternalManagerProvider({ children }: PropsWithChildren) {
   const [transactionStatus, setTransactionStatus] = useState<TransactionStatus>(
     TransactionStatus.AUTHORIZING
   );
+  const [tokenList, setTokenList] = useState<tokenListItem[]>([]);
+
+  const { networkConnectedToWallet } = useNetworks();
 
   function clearNotification() {
     setNotification(null);
     setIsNotifierOpen(false);
   }
 
+  useEffect(() => {
+    async function getTokensData() {
```

Comment:
as you mentioned in the video I would avoid fetching the data on every network change